### PR TITLE
[Sync] Enqueue background survey sync on messages from FCM

### PR DESCRIPTION
**Pull request by gino-m**
Towards #1373. 

Depends on #1360.

### CODE REVIEW

**Comment by gino-m** [comment]
@shobhitagarwal1612 can you PTAL after merging #1360? Be sure to "Update branch" first so that changes from #1360 don't appear in the diff.

**Comment by shobhitagarwal1612** [comment]
Will TAL once changes are rebased/merged with HEAD

**Comment by gino-m** [comment]
> Will TAL once changes are rebased/merged with HEAD

@shobhitagarwal1612 Sorry, I thought GitHub would allow you to merge interactively. Pushed a merge commit, RFAL.

File: ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirestoreDataStore.kt [hunk 1]
```diff
@@ -140,6 +142,12 @@ internal constructor(
       .subscribeOn(schedulers.io())
   }
 
+  override fun subscribeToSurveyUpdates(surveyId: String): Completable =
+    RxTask.toCompletable {
+      Timber.d("Subscribing to FCM topic $surveyId")
+      Firebase.messaging.subscribeToTopic(surveyId)
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirestoreDataStore.kt:148]: What happens if we switch the survey? Does the old subscription get expired or will it still be active?

**Review by shobhitagarwal1612 — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirestoreDataStore.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirestoreDataStore.kt:148]: For now it stays active. To fix this, we unsubscribe when the user turns off "available offline" for synced surveys as part of #450.

**Comment by shobhitagarwal1612** [comment]
> I thought GitHub would allow you to merge interactively. Pushed a merge commit, RFAL.

I've enabled the setting. Github should now suggest updating the branch.

**Pull request merged**